File: repos_estudiantes_ALVAREZ_tercera_entrega/ALBA_PLATA_BRYAN_ALBERTO/PracticaTRES/GENERECIDAD/ejercicio5/Pila.py
class Pila:
    def __init__(self):
        self.items = []

    def apilar(self, item):
        self.items.append(item)

    def desapilar(self):
        return self.items.pop() if self.items else None

    def mostrar(self):
        print("Contenido de la pila:", self.items)


# Prueba con enteros
pila_enteros = Pila()
pila_enteros.apilar(1)
pila_enteros.apilar(2)
pila_enteros.apilar(3)
pila_enteros.mostrar()
pila_enteros.desapilar()
pila_enteros.mostrar()

# Prueba con strings
pila_textos = Pila()
pila_textos.apilar("uno")
pila_textos.apilar("dos")
pila_textos.mostrar()

File: repos_estudiantes_ALVAREZ_tercera_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/GENERACIDAD/Ejercicio5/Pila.java

//Ejercicio 5: Crea una clase genérica Pila<T> 
import java.util.ArrayList;

public class Pila<T> {
    private ArrayList<T> elementos = new ArrayList<>();

    // a) Implementa un método para apilar
    public void apilar(T dato) {
        elementos.add(dato);
    }
    // b) Implementa un método para des apilar

    public T desapilar() {
        if (!elementos.isEmpty()) {
            return elementos.remove(elementos.size() - 1);
        }
        return null;
    }

    public void mostrar() {
        System.out.println("Contenido de la pila:");
        for (int i = elementos.size() - 1; i >= 0; i--) {
            System.out.println(elementos.get(i));
        }
    }

    public static void main(String[] args) {
        // c) Prueba la pila con diferentes tipos de datos
        // d) Muestra los datos de la pila
        Pila<String> pilaTexto = new Pila<>();
        pilaTexto.apilar("uno");
        pilaTexto.apilar("dos");
        pilaTexto.mostrar();
        System.out.println("Desapilado: " + pilaTexto.desapilar());
        pilaTexto.mostrar();

        Pila<Integer> pilaNumeros = new Pila<>();
        pilaNumeros.apilar(451);
        pilaNumeros.apilar(2222);
        pilaNumeros.mostrar();
        System.out.println("Desapilado: " + pilaNumeros.desapilar());
        pilaNumeros.mostrar();
    }
}
